python/ql/src/semmle/python/essa/SsaCompute.qll

```diff
@@ -382,7 +382,9 @@ private module SsaComputeImpl {
 
     /** Holds if `v` occurs in `b` or one of `b`'s transitive successors. */
     private predicate blockPrecedesVar(SsaSourceVariable v, BasicBlock b) {
-      varOccursInBlock(v, b.getASuccessor*())
+      varOccursInBlock(v, b)
+      or
+      SsaDefinitionsImpl::reachesEndOfBlock(v, _, _, b)
```

Comment:
Could this simply be `SsaDefinitionsImpl::reachesEndOfBlock(v, _, _, b)`?

Reply:
I don't think so, since `reachesEndOfBlock` requires that `v` is live at exit ([src](https://github.com/github/codeql/blob/71da9045e5361c261c1acb5af5ddb7bdfbcbd828/python/ql/src/semmle/python/essa/SsaCompute.qll#L311)) -- so `reachesEndOfBlock` would not hold for a use in the last block where `v` is used. (that's at least according to my logic, I did not run any QL to verify my logic holds)

Reply:
Correct @RasmusWL .

Reply:
Ah, makes sense.